### PR TITLE
Fixes #1712 Plugin Download and Install from Market

### LINKED ISSUE #1712: Unable to install plugins or themes via admin web interface....
I'm having a "Can't continue" error when i try to install plugins or themes via the web interface on the admin area....
Any ideas what might be causing it???

### PR DESCRIPTION
**Pull request by MikeyGMT**
Fixes #1712 Plugin Download and Install from Market - Plus added debug statements around plugin download. Tested on 3 sites.

### CODE REVIEW

File: e107_admin/plugin.php [hunk 1]
```diff
@@ -665,51 +680,59 @@ private function pluginDownload()
...
+		$this->id         = !empty($tmp[1]) 	? intval($tmp[1]) 	: "";//mikeyfixed
+		$this->id         =  empty($this->id) 	? $plugin_id 		: $this -> id;//mikeyfixed
 	//	print_a($_GET); 	
-		
 		$string =  base64_decode($_GET['src']);	
+
 		parse_str($string, $data);
 
-		if(!empty($data['plugin_price']))
+		if(!empty($plugin_price))
```
Comment (CaMer0n) [inline e107_admin/plugin.php]: Why not use $data, is it empty?

File: e107_handlers/e_marketplace.php [hunk 2]
```diff
@@ -492,26 +538,30 @@ class e_marketplace_adapter_wsdl extends e_marketplace_adapter_abstract
...
-			'exception' 			=> true,
-		    "uri" 					=> "http://server.soap.e107.inc.com/",
-		    'cache_wsdl'			=> WSDL_CACHE_NONE,
-		    'connection_timeout' 	=> 60,
+			"trace"					=> true, 
+			'exception'				=> true,
+			"uri"					=> "http://server.soap.e107.inc.com/",
+			'cache_wsdl'			=> WSDL_CACHE_NONE,
+			'connection_timeout'	=> 60,
 		);
```
Comment (CaMer0n) [inline e107_handlers/e_marketplace.php]: Your tabs/spaces are showing as changes. Makes it difficult to track what you changed. :-(

**Comment by CaMer0n** [comment]
@MikeyGMT Thank you. I'd prefer not to clutter the debug log if possible. Was the issue the lack of curly brackets on the `else `?

File: e107_admin/plugin.php [hunk 3]
```diff
@@ -665,51 +680,59 @@ private function pluginDownload()
 		$frm = e107::getForm();
 		$mes = e107::getMessage();
 		
+		e107::getDebug()->log("Plugin Download.");
+		$tmp = explode('&amp;', e_QUERY); // #1712 we need to process new ampersand separated query
```
Comment (MikeyGMT) [inline e107_admin/plugin.php]: Best way to learn is have a go. 
What I found was $_GET only contained mode=download, there was no other data in there.
src is encrypted and in e_QUERY and that $tmp was already trying to grab values from exploding e_QUERY using .dot. separator. 
However, we got &amp; separators. So I added check to see id e_query contained &'s then exploded with that.

File: e107_admin/plugin.php (hunk 1 above)
Comment (MikeyGMT) [inline e107_admin/plugin.php]: Yup data was empty ...

File: e107_admin/plugin.php (hunk 3 above)
Comment (MikeyGMT) [inline e107_admin/plugin.php]: I was unsure about leaving debug calls - they only show up in debug mode... when we need them. Agree it adds clutter. Will trim them down. 

Good learning experience. I suppose, why was $_GET and $data are empty may be a better fix ? I'm online today. Busy this evening and weekend.

File: e107_admin/plugin.php (hunk 1 above)
Comment (MikeyGMT) [inline e107_admin/plugin.php]: Check details here... 
 https://docs.google.com/document/d/1hbq645DZHW2Zy1oSos71yJMiqHbEk06BObM5HvzWNgQ

I'm back after the weekend :)

File: e107_admin/plugin.php (hunk 3 above)
Comment (CaMer0n) [inline e107_admin/plugin.php]: I can't reproduce any of these issues on my local machine. If you can grant me FTP access to a machine that has this issue, I could run some tests. I have a hunch it is the use of '&amp' in the URL. I will commit that change and we can see how that affects things.

**Comment by CaMer0n** [comment]
@MikeyGMT Do I understand that this issue is resolved?

**Pull request closed without merging**